app/models/petition.rb

```diff
@@ -28,6 +28,7 @@ class Petition < ActiveRecord::Base
 
   has_perishable_token called: 'sponsor_token'
 
+  before_save :stamp_parliament_response_at, if: -> { response.present? && response_was.nil? }
   after_create :set_petition_on_creator_signature
```

Comment:
@davidbasalla `response_was` (or "#{attr}_was") is an `ActiveRecord` feature, which is useful when you want to check if the value has been changed. All model attributes have these methods. Some examples: https://github.com/rails/rails/blob/3e36db4406beea32772b1db1e9a16cc1e8aea14c/activemodel/lib/active_model/dirty.rb